packages/opentelemetry-node/src/instrumentation/utils.ts

```diff
@@ -72,3 +72,35 @@ export function isSupportedVersion(
 export function searchPathForTest(searchPath: string) {
   module.paths.push(searchPath);
 }
+
+/**
+ * Merge plugin options. Rightmost options take precedence.
+ *
+ * @param options config objects to merge
+ */
+export function mergeOptions(
```

Comment:
Missing unit tests.
This function doesn't guarantee the "frozen state" when option is array or object.

Reply:
I'll take another pass at this